x-pack/plugins/ecs_data_quality_dashboard/server/schemas/result.ts

```diff
@@ -19,8 +19,25 @@ const ResultDocumentInterface = t.interface({
   ecsFieldCount: t.number,
   customFieldCount: t.number,
   incompatibleFieldCount: t.number,
+  incompatibleFieldItems: t.array(
+    t.type({
+      fieldName: t.string,
+      expectedValue: t.union([t.string, t.array(t.string)]),
+      actualValue: t.union([t.string, t.array(t.string)]),
```

Comment:
Consider storing the counts of the incompatible values (in addition to the incompatible values) to make it possible to render (from stored results) the counts shown in the screenshot below:

![count_of_actual_value](https://github.com/elastic/kibana/assets/4459398/a0e1df6f-17a1-474c-9906-a6f820302002)

To create the screenshot above, a 2nd document containing the value was inserted (via dev tools): 

```
PUT test-field-items/_doc/2
{
  "@timestamp": "2024-06-11T08:05:34.853Z",
  "event.category": "behavior"
}
```

The count is already (correctly) included in the `markdownComments` field in stored results: 

```json
                {
                  "_index": ".ds-.kibana-data-quality-dashboard-results-default-2024.05.22-000001",
                  "_id": "PuIGCZABPzDWWqs-bmTR",
                  "_score": null,
                  "_source": {
                    "batchId": "a650d1f6-c0d7-400a-b023-23edd23e27af",
                    "indexName": "test-field-items",
                    "indexPattern": "test-*",
                    "isCheckAll": false,
                    "checkedAt": 1718138334792,
                    "docsCount": 1,
                    "totalFieldCount": 4,
                    "ecsFieldCount": 1,
                    "customFieldCount": 0,
                    "incompatibleFieldCount": 2,
                    "incompatibleFieldItems": [
                      {
                        "fieldName": "event.category",
                        "expectedValue": [
                          "api",
                          "authentication",
                          "configuration",
                          "database",
                          "driver",
                          "email",
                          "file",
                          "host",
                          "iam",
                          "intrusion_detection",
                          "library",
                          "malware",
                          "network",
                          "package",
                          "process",
                          "registry",
                          "session",
                          "threat",
                          "vulnerability",
                          "web"
                        ],
                        "actualValue": [
                          "behavior"
                        ],
                        "description": """This is one of four ECS Categorization Fields, and indicates the second level in the ECS category hierarchy.
`event.category` represents the "big buckets" of ECS categories. For example, filtering on `event.category:process` yields all events relating to process activity. This field is closely related to `event.type`, which is used as a subcategory.
This field is an array. This will allow proper categorization of some events that fall in multiple categories.""",
                        "reason": "value"
                      },
                      {
                        "fieldName": "source.ip",
                        "expectedValue": "ip",
                        "actualValue": "text",
                        "description": "IP address of the source (IPv4 or IPv6).",
                        "reason": "mapping"
                      }
                    ],
                    "sameFamilyFieldCount": 1,
                    "sameFamilyFields": [
                      "agent.type"
                    ],
                    "sameFamilyFieldItems": [
                      {
                        "fieldName": "agent.type",
                        "expectedValue": "keyword",
                        "actualValue": "constant_keyword",
                        "description": """Type of the agent.
The agent type always stays the same and should be given by the agent used. In case of Filebeat the agent would always be Filebeat also if two Filebeat instances are run on the same machine."""
                      }
                    ],
                    "unallowedMappingFields": [
                      "source.ip"
                    ],
                    "unallowedValueFields": [
                      "event.category"
                    ],
                    "sizeInBytes": 8780,
                    "ilmPhase": "unmanaged",
                    "markdownComments": [
                      """### test-field-items
""",
                      """| Result | Index | Docs | Incompatible fields | ILM Phase | Size |
|--------|-------|------|---------------------|-----------|------|
| ❌ | test-field-items | 1 (100.0%) | 2 | `unmanaged` | 8.6KB |

""",
                      """### **Incompatible fields** `2` **Same family** `1` **Custom fields** `0` **ECS compliant fields** `1` **All fields** `4`
""",
                      """#### 2 incompatible fields

Fields are incompatible with ECS when index mappings, or the values of the fields in the index, don't conform to the Elastic Common Schema (ECS), version 8.11.0.

❌ Detection engine rules referencing these fields may not match them correctly
❌ Pages may not display some events or fields due to unexpected field mappings or values
❌ Mappings or field values that don't comply with ECS are not supported
""",
                      """
#### Incompatible field mappings - test-field-items


| Field | ECS mapping type (expected) | Index mapping type (actual) | 
|-------|-----------------------------|-----------------------------|
| source.ip | `ip` | `text`  |

#### Incompatible field values - test-field-items


| Field | ECS values (expected) | Document values (actual) | 
|-------|-----------------------|--------------------------|
| event.category | `api`, `authentication`, `configuration`, `database`, `driver`, `email`, `file`, `host`, `iam`, `intrusion_detection`, `library`, `malware`, `network`, `package`, `process`, `registry`, `session`, `threat`, `vulnerability`, `web` | `behavior` (2) |

"""
                    ],
                    "ecsVersion": "8.11.0",
                    "indexId": "NoMNexi_Th-duyhPmnDmlA",
                    "error": null,
                    "@timestamp": 1718138334831,
                    "checkedBy": "u_mGBROF_q5bmFCATbLXAcCwKa0k8JvONAwSruelyKA5E_0"
                  },
                  "sort": [
                    1718138334831
                  ]
                }
```

Reply:
Good catch @andrew-goldstein, we did not realize this counter existed. 
@kapral18 what do you think about adding `actualValueCount` inside the `incompatibleFieldItems`? I assume it's also necessary for `sameFamilyFieldItems` as well.

Reply:
@andrew-goldstein I knew the wait was worth the while, thanks for finding this.

@semd 2 things to keep in mind:

1. SameFamily fields are valid so cannot have actual value counts, they only show up if there is a type mismatch within acceptable match, so I guess we don't need to have any of that in sameFamilyFieldItems
![image](https://github.com/elastic/kibana/assets/1625373/09b4f77e-a8f3-40a1-be80-af85fb177d7a)

2. Since we can have multiple actualValues for the same field 
![image](https://github.com/elastic/kibana/assets/1625373/26ce6ccf-45e6-4644-8d47-ea9368ed3c1f)
I'd argue it's a little more convenient to include the count into `actualValue` list but now I am also thinking about tweaking the shape a little more to make it more evident that it's an actual list in case of incompatible items vs same family field to avoid confusions like this.

I will push the change and it will be easier to see the suggestion in the separate commit.

Reply:
@kapral18 correct, "same family" is always a mapping issue, they don't have a value appearance counter. Same for "mapping incompatibilities", this counter is only needed for "value incompatibilities". This will need to be stored per-"incompatible actual value".

Reply:
@andrew-goldstein @semd @angorayc after weighing the options I decided to revamp the underlying structure one last time. I believe it's much more explicit now. Please have a look.

The gist of changes I separate incompatible mappings and values from each other and values now contain `actualValues` instead of `actualValue` and include the `count` with each invalid value item

Reply:
I also updated the PR description